src/elements/common/collaborators/withCollaborators.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import * as React from 'react';
+import debounce from 'lodash/debounce';
+import { DEFAULT_COLLAB_DEBOUNCE } from '../../../constants';
+import API from '../../../api/APIFactory';
+import type { Collaborators } from '../../content-sidebar/flowTypes';
+import type { ElementsErrorCallback } from '../../../common/types/api';
+
+type Props = {
+    api: API,
+};
+
+export default function withCollaborators(WrappedComponent: React.ComponentType<Props>): React.ComponentType<Props> {
+    class WithMentions extends React.Component<Props> {
+        static displayName: ?string;
+
+        props: Props;
+
+        /**
+         * Fetches file collaborators
+         *
+         * @param {string} fileId
+         * @param {Function} successCallback - the success callback
+         * @param {Function} errorCallback - the error callback
+         * @param {string} searchStr - the search string
+         * @param {Object} [options]
+         * @param {boolean} [options.includeGroups] - return groups as well as users
+         * @return {void}
+         */
+        getCollaborators = (
```

Comment:
If this is a commonly-needed behavior, does it need to live in a separate HOC or can we augment the `FileCollaborators` API module, instead?

Reply:
I think HOC helps with handling additional debounce logic. If you don't mind, I would rather to keep it that way